### PR TITLE
Adds support for new currencies and tidies up some currency related code

### PR DESCRIPTION
**Pull request by jonrandy**
#### 1. Objective

Adds support for new currencies recently added on the backend:

```
TH:

Australian Dollar (AUD)
Swiss Franc (CHF)
Chinese Yuan (CNY)
Danish Krone (DKK)
Euro (EUR)
British Pound (GBP)
Hongkong Dollar (HKD)
Japanese Yen (JPY)
Singapore Dollar (SGD)
US Dollar (USD)

SG:

Australian Dollar (AUD)
Canadian Dollar (CAD)
Swiss Franc (CHF)
China Yuan (CNY)
Euro (EUR)
British Pound (GBP)
Hongkong Dollar (HKD)
Japanese Yen (JPY)
Malaysian Ringgit (MYR)
Thai Baht (THB)
US Dollar (USD)
```

#### 2. Description of change

Reworked code relating to currencies + fixed some code that was unnecessarily formatting prices so that it uses Magento's built in formatting functions instead

#### 3. Quality assurance

Tested on Magento 1.9 - made sure all payments in supported currencies working and that amount formatting on the 'Omise dashboard' admin page is working correctly

#### 4. Impact of the change

New currencies can now be used

#### 5. Priority of change

Normal

#### 6. Additional Notes

**🎶**

### CODE REVIEW

**Comment by jacstn** [comment]
tested, works good.

File: src/app/code/community/Omise/Gateway/Helper/Data.php [hunk 1]
```diff
@@ -1,30 +1,107 @@
 <?php
 class Omise_Gateway_Helper_Data extends Mage_Core_Helper_Abstract
 {
-    public function formatPrice($currency, $amount)
-    {
-
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Helper/Data.php]: this space not necessary probably.

File: src/app/code/community/Omise/Gateway/Model/Payment.php [hunk 2]
```diff
@@ -6,17 +6,18 @@ abstract class Omise_Gateway_Model_Payment extends Mage_Payment_Model_Method_Abs
...
-     */
-    private $_currency_subunits = array(
-        'JPY' => 1,
-        'THB' => 100,
-        'SGD' => 100,
-        'USD' => 100,
-        'EUR' => 100,
-        'GBP' => 100
-    );
+    // /**
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Model/Payment.php]: please remove that comment

File: src/app/code/community/Omise/Gateway/Model/Payment.php [hunk 3]
```diff
@@ -36,26 +37,31 @@ public function __construct()
      */
     protected function _isCurrencySupport($currency)
     {
-        if (isset($this->_currency_subunits[strtoupper($currency)])) {
-            return true;
-        }
+        return Mage::helper('omise_gateway')->isCurrencySupported($currency);
 
-        return false;
+        // if (isset($this->_currency_subunits[strtoupper($currency)])) {
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Model/Payment.php]: please remove this comment block.

File: src/app/code/community/Omise/Gateway/Model/Payment.php [hunk 4]
```diff
@@ -36,26 +37,31 @@ public function __construct()
...
      * @param  string $currency
      *
      * @return int
      */
     public function getAmountInSubunits($amount, $currency)
     {
-        if ($this->_isCurrencySupport($currency)) {
-            return $this->_currency_subunits[$currency] * $amount;
-        }
+        // ////// move to omise_gateway helper
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Model/Payment.php]: please remove this comment block

File: src/app/code/community/Omise/Gateway/Model/Payment.php (hunk 2 above)
Comment (jonrandy) [reply src/app/code/community/Omise/Gateway/Model/Payment.php]: 5555... I already removed all these. Just looked at editor - I hadn't clicked save!! :P

File: src/app/code/community/Omise/Gateway/Helper/Data.php [hunk 5]
```diff
@@ -1,30 +1,107 @@
...
+     *
+     * @param string $currency
+     * @return bool
+     */
+    public function isCurrencySupported($currency)
+    {
+        return in_array(strtoupper($currency), self::$supportedCurrencies);
+    }
+
+
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Helper/Data.php]: still unnecessary double space

File: src/app/code/community/Omise/Gateway/Helper/Data.php [hunk 6]
```diff
@@ -1,30 +1,107 @@
...
+     * @param string $currency
+     * @param numeric $amount
+     * @return string
+     */
+    public function subunitsToAmount($currency, $subunits)
+    {
+        return $subunits / self::subUnitsFor($currency);
+    }
+
+
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Helper/Data.php]: double space, unnecessary probably

File: src/app/code/community/Omise/Gateway/Helper/Data.php [hunk 7]
```diff
@@ -1,30 +1,107 @@
...
+            'HKD',
+            'JPY',
+            'MYR',
+            'THB',
+            'SGD',
+            'USD'
+        ]
+    ;
+
+
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Helper/Data.php]: double space unnecessary probably.

File: src/app/code/community/Omise/Gateway/Helper/Data.php [hunk 8]
```diff
@@ -1,30 +1,107 @@
...
-                }
-                break;
-
-            case 'JPY':
-                $amount = number_format($amount) . "円";
-                break;
-        }
-
-        return $amount;
+
```
Comment (jacstn) [inline src/app/code/community/Omise/Gateway/Helper/Data.php]: this spac probably unnecessary

**Pull request merged**